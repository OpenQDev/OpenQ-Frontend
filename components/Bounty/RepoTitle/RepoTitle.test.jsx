/**
 * @vi-environment jsdom
 */
import React from 'react';

import RepoTitle from '.';
import ShallowRenderer from 'react-test-renderer/shallow';

// WARNING If you change the mock data for issues you may need to change some
// of this test's getByText invocations to getAllByText.
describe('RepoTitle', () => {
  it('should render BountyDetails and match DOM Snapshot', () => {
    const shallow = new ShallowRenderer();
    shallow.render(<RepoTitle repo={{ owner: 'testing', name: 'testing' }} />);
    const tree = shallow.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });
});
