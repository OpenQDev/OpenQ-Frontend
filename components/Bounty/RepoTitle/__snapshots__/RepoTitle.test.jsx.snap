// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`RepoTitle > should render BountyDetails and match DOM Snapshot 1`] = `
<div
  className="flex items-center justify-between pb-5 w-full px-2 sm:px-8"
>
  <div
    className="flex items-center gap-2"
  >
    <svg
      className="fill-muted"
      height="16"
      viewBox="0 0 16 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M2 2.5A2.5 2.5 0 014.5 0h8.75a.75.75 0 01.75.75v12.5a.75.75 0 01-.75.75h-2.5a.75.75 0 110-1.5h1.75v-2h-8a1 1 0 00-.714 1.7.75.75 0 01-1.072 1.05A2.495 2.495 0 012 11.5v-9zm10.5-1V9h-8c-.356 0-.694.074-1 .208V2.5a1 1 0 011-1h8zM5 12.25v3.25a.25.25 0 00.4.2l1.45-1.087a.25.25 0 01.3 0L8.6 15.7a.25.25 0 00.4-.2v-3.25a.25.25 0 00-.25-.25h-3.5a.25.25 0 00-.25.25z"
        fillRule="evenodd"
      />
    </svg>
    <div
      className="text-xl"
    >
      <span>
        <ForwardRef(LinkComponent)
          data-testid="repo"
          href="undefined/organization/OpenQDev"
        >
          <span
            className="text-link-colour hover:underline cursor-pointer"
          >
            OpenQDev
          </span>
        </ForwardRef(LinkComponent)>
         / 
        <ForwardRef(LinkComponent)
          data-testid="repo"
          href="undefined/repo/OpenQDev/OpenQ-Frontend"
        >
          <span
            className="text-link-colour hover:underline cursor-pointer"
          >
            OpenQ-Frontend
          </span>
        </ForwardRef(LinkComponent)>
      </span>
    </div>
  </div>
  <WatchButton
    bounty={
      {
        "address": "0x066efd87027a4cc4a21c57467f224ef98241d774",
        "assignees": [],
        "avatarUrl": "https://avatars.githubusercontent.com/u/77402538?v=4",
        "body": "body of test2",
        "bodyHTML": "<p dir=\\"auto\\">body of test2</p>",
        "bountyAddress": "0x066efd87027a4cc4a21c57467f224ef98241d774",
        "bountyId": "I_kwDOGWnnz85I9Agl",
        "bountyMintTime": "1661785920",
        "bountyTokenBalances": [
          {
            "__typename": "BountyFundedTokenBalance",
            "tokenAddress": "0x0000000000000000000000000000000000000000",
            "volume": "2000000000000000000",
          },
          {
            "__typename": "BountyFundedTokenBalance",
            "tokenAddress": "0x9fE46736679d2D9a65F0992F2272dE9f3c7fa6e0",
            "volume": "2000000000000000000",
          },
          {
            "__typename": "BountyFundedTokenBalance",
            "tokenAddress": "0x5FbDB2315678afecb367f032d93F642f64180aa3",
            "volume": "2000000000000000000",
          },
        ],
        "bountyType": "0",
        "claims": [],
        "closed": true,
        "closedAt": "2022-03-28T17:57:44Z",
        "closedEvents": [
          {
            "__typename": "ClosedEvent",
            "actor": {
              "__typename": "User",
              "avatarUrl": "https://avatars.githubusercontent.com/u/93455288?u=fd1fb04b6ff2bf397f8353eafffc3bfb4bd66e84&v=4",
              "login": "FlacoJones",
              "name": null,
              "url": "https://github.com/FlacoJones",
            },
            "id": "CE_lADOGWnnz85GjwA1zwAAAAF4vHFc",
          },
        ],
        "closerData": null,
        "createdAt": "1661767948472",
        "deposits": [
          {
            "__typename": "Deposit",
            "expiration": "0",
            "id": "0xe5551a3fa87d93a0c6c084d572b9e282114befc43dc68f08be6d53d13830e356",
            "receiveTime": "1661785920",
            "refundTime": null,
            "refunded": false,
            "sender": {
              "__typename": "User",
              "id": "0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266",
            },
            "tokenAddress": "0x0000000000000000000000000000000000000000",
            "volume": "1000000000000000000",
          },
          {
            "__typename": "Deposit",
            "expiration": "0",
            "id": "0xb4f31aab8a1c4bfe26236729e8cd8e4abf81d63283e006b4ec677a7ce6b2871a",
            "receiveTime": "1661785920",
            "refundTime": "1661785920",
            "refunded": true,
            "sender": {
              "__typename": "User",
              "id": "0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266",
            },
            "tokenAddress": "0x9fE46736679d2D9a65F0992F2272dE9f3c7fa6e0",
            "volume": "2000000000000000000",
          },
          {
            "__typename": "Deposit",
            "expiration": "0",
            "id": "0x7db2691e573b9c19e90f391cd3eda9ce9246666a1502f2bf87b9d47d13679bc0",
            "receiveTime": "1661785920",
            "refundTime": null,
            "refunded": false,
            "sender": {
              "__typename": "User",
              "id": "0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266",
            },
            "tokenAddress": "0x5fbdb2315678afecb367f032d93f642f64180aa3",
            "volume": "1000000000000000000",
          },
        ],
        "fundingGoalTokenAddress": "0x0000000000000000000000000000000000000000",
        "id": undefined,
        "issuer": {
          "__typename": "User",
          "id": "0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266",
        },
        "labels": [
          {
            "__typename": "Label",
            "color": "0075ca",
            "name": "documentation",
          },
          {
            "__typename": "Label",
            "color": "cfd3d7",
            "name": "duplicate",
          },
        ],
        "number": 8,
        "owner": "OpenQDev",
        "prs": [
          {
            "__typename": "CrossReferencedEvent",
            "assignees": [],
            "id": "CRE_kwDOGWnnz85GHUw1",
            "referencedAt": "2022-03-28T17:47:26Z",
            "source": {
              "__typename": "PullRequest",
              "author": {
                "__typename": "User",
                "avatarUrl": "https://avatars.githubusercontent.com/u/93455288?u=fd1fb04b6ff2bf397f8353eafffc3bfb4bd66e84&v=4",
                "email": "",
                "id": "U_kgDOBZIDuA",
                "login": "FlacoJones",
                "url": "https://github.com/FlacoJones",
                "user": {
                  "__typename": "User",
                  "login": "FlacoJones",
                  "url": "https://github.com/FlacoJones",
                },
              },
              "body": "This Closes #136 and also unrelated thing of Fixes #137",
              "bodyHTML": "<p dir=\\"auto\\">This <span class=\\"issue-keyword tooltipped tooltipped-se\\" aria-label=\\"This pull request closes issue #136.\\">Closes</span> <a class=\\"issue-link js-issue-link\\" data-error-text=\\"Failed to load title\\" data-id=\\"1183776821\\" data-permission-text=\\"Title is private\\" data-url=\\"https://github.com/OpenQDev/OpenQ-TestRepo/issues/136\\" data-hovercard-type=\\"issue\\" data-hovercard-url=\\"/OpenQDev/OpenQ-TestRepo/issues/136/hovercard\\" href=\\"https://github.com/OpenQDev/OpenQ-TestRepo/issues/136\\">#136</a> and also unrelated thing of <span class=\\"issue-keyword tooltipped tooltipped-se\\" aria-label=\\"This pull request closes issue #137.\\">Fixes</span> <a class=\\"issue-link js-issue-link\\" data-error-text=\\"Failed to load title\\" data-id=\\"1183777365\\" data-permission-text=\\"Title is private\\" data-url=\\"https://github.com/OpenQDev/OpenQ-TestRepo/issues/137\\" data-hovercard-type=\\"issue\\" data-hovercard-url=\\"/OpenQDev/OpenQ-TestRepo/issues/137/hovercard\\" href=\\"https://github.com/OpenQDev/OpenQ-TestRepo/issues/137\\">#137</a></p>",
              "bodyText": "This Closes #136 and also unrelated thing of Fixes #137",
              "createdAt": "2022-03-28T17:57:44Z",
              "id": "PR_kwDOGWnnz841LGsK",
              "mergeCommit": {
                "__typename": "Commit",
                "author": {
                  "__typename": "GitActor",
                  "avatarUrl": "https://avatars.githubusercontent.com/u/93455288?v=4",
                  "name": "FlacoJones",
                  "user": {
                    "__typename": "User",
                    "login": "FlacoJones",
                    "url": "https://github.com/FlacoJones",
                  },
                },
              },
              "merged": true,
              "mergedAt": "2022-03-28T17:57:44Z",
              "title": "Update README.md",
              "url": "https://github.com/OpenQDev/OpenQ-TestRepo/pull/138",
            },
          },
        ],
        "refunds": [
          {
            "__typename": "Refund",
            "depositId": "0xb4f31aab8a1c4bfe26236729e8cd8e4abf81d63283e006b4ec677a7ce6b2871a",
            "refundTime": "1661785920",
            "tokenAddress": "0x9fE46736679d2D9a65F0992F2272dE9f3c7fa6e0",
            "volume": "2000000000000000000",
          },
        ],
        "repoName": "OpenQ-Frontend",
        "title": "Properly Referenced and Merged by FlacoJones",
        "titleHTML": "Properly Referenced and Merged by FlacoJones",
        "tvl": "9.51848",
        "twitterUsername": "openqlabs",
        "url": "https://github.com/OpenQDev/OpenQ-TestRepo/issues/8",
      }
    }
  />
</div>
`;
